azure/network.tf

```diff
@@ -25,10 +25,24 @@ resource "azurerm_proximity_placement_group" "redpanda" {
 }
 
 resource "azurerm_availability_set" "redpanda" {
-  name                         = "redpanda_availability_set"
+  name                         = "redpanda_availability_set_${local.deployment_id}"
   resource_group_name          = azurerm_resource_group.redpanda.name
   location                     = azurerm_resource_group.redpanda.location
   proximity_placement_group_id = azurerm_proximity_placement_group.redpanda.id
+  count                        = var.ha ? 0 : 1
```

Comment:
Same questions as above: this value seems opposite to what I would expect we would want (to create an availability set when `ha` is `true`).